framework/audio/internal/platform/win/winmmdriver.cpp

```diff
@@ -0,0 +1,215 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "winmmdriver.h"
+
+#include <windows.h>
+#include <mmsystem.h>
+
+#ifdef _MSC_VER
+#pragma comment(lib, "winmm.lib")
+#endif
+
+#include "log.h"
+
+using namespace mu::audio;
+
+namespace  {
+static const int BUFFER_COUNT = 4;
+
+struct WinMMData
+{
+    short* sampleBuffer[BUFFER_COUNT];

```

Comment:
it's an array of pointers on buffers. I'd suggest to rename it to sampleBufferPtr